### PR TITLE
Add REP process definition and template

### PR DESCRIPTION
**Pull request by shenglol**
- Updated README.md to define the Bicep REP workflow
- Added a REP template

References:
- https://github.com/PowerShell/PowerShell-RFC
- https://github.com/rust-lang/rfcs
- https://github.com/reactjs/rfcs

### CODE REVIEW

File: 0000-rep-template.md
Comment (StephenWeatherford) [inline 0000-rep-template.md:3]: not github username?

File: 0000-rep-template.md
Comment (majastrz) [inline 0000-rep-template.md:37]: Should we have some sort of title modifier or text or tag somewhere that makes it more obvious that we're dealing with a breaking changes REP?

File: 0000-rep-template.md
Comment (majastrz) [inline 0000-rep-template.md:5]: Could be multiple issues in some cases. Maybe we just need to add " (s)" to the title?

File: README.md
Comment (alex-frankel) [inline README.md:8]: "feature" is pretty broad - do we specifically mean a syntax change?

File: README.md
Comment (majastrz) [reply README.md:33]: Also do we want to have a more light-weight process for user suggestions before (or if) they turn into REPs?

File: README.md
Comment (alex-frankel) [inline README.md:18]: should we be more specific thatm in most cases, starting with an issue for a feature proposal is more appropriate?

File: 0000-rep-template.md
Comment (shenglol) [reply 0000-rep-template.md:3]: Using both makes sense.

File: 0000-rep-template.md
Comment (asilverman) [inline 0000-rep-template.md:38]: Should we add an Out of Scope:
> [Out of scope](https://gist.github.com/rynowak/5a39b92997546216f957f0aa2dc4b32b#out-of-scope)
Describe the important issues or concerns NOT addressed by this proposal. 1 paragraph each (can be short)

Often the "out of scope" items are refer to the current state. We might already have planned work to address the "out of scope" items, mention it if that's the case".

File: README.md
Comment (shenglol) [inline README.md:46]: Make it clear that all unresolved questions are resolved.

File: README.md
Comment (StephenWeatherford) [inline README.md:23]: Code of conduct section might need to be here still...?

File: README.md
Comment (majastrz) [inline README.md:76]: Do we need an Experimental REP state similar to how PowerShell RFCs handle experimental features?

File: README.md
Comment (StephenWeatherford) [inline README.md:27]: ditto

File: 0000-rep-template.md
Comment (StephenWeatherford) [inline 0000-rep-template.md:2]:
```suggestion
REP Number: 0000 <Replace me in with the pull request number update AFTER REP acceptance.>
```

Comment (StephenWeatherford) [inline 0000-rep-template.md:2]: This is confusing.  This should be the pull request number?  If so, "unique incrementing number" doesn't make sense.  So this should be left blank until it's accepted?  Can a REP move from draft immediately to inactive?

File: 0000-rep-template.md
Comment (shenglol) [inline 0000-rep-template.md:20]: Add [Terms and definitions](https://github.com/radius-project/design-notes/blob/main/template/YYYY-MM-design-template.md#terms-and-definitions).

File: README.md
Comment (StephenWeatherford) [inline README.md:80]: Are there times we might want to keep the rejected proposals for records?  Rejecting and deleting the PR and branch are permanent.  Maybe optionally move to a rejected folder?

File: README.md
Comment (shenglol) [reply README.md:33]: Suggested final approvers: @majastrz @anthony-c-martin @alex-frankel .

Comment (shenglol) [reply README.md:33]: We can use Bicep discussion meetings to review REPs.

Comment (shenglol) [reply README.md:33]: Number of approvals does not matter.

File: README.md
Comment (shenglol) [inline README.md:51]: Inactive => Deferred

File: README.md
Comment (shenglol) [reply README.md:80]: Based on the discussion, we'd like to use closed PRs to track them to reduce mantainance overheads.

File: README.md
Comment (shenglol) [reply README.md:76]: My expectation is that most REP features are likely to be initially implemented as experimental, so I'm not sure if it's worth maintaining two active states, one for non-experimental features and one for experimental features. What do you think?

File: 0000-rep-template.md
Comment (shenglol) [reply 0000-rep-template.md:2]: PR numbers are unique incrementing numbers, but I guess it is a bit confusing to explicitly mention it here. The idea is to left it blank until it's accepted and before merged. Will update this to reflect that.

Comment (shenglol) [reply 0000-rep-template.md:2]: > Can a REP move from draft immediately to inactive?

No, only a PR that tries to add a REP file to the `/active` folder is considered a draft.

File: 0000-rep-template.md
Comment (shenglol) [reply 0000-rep-template.md]: Update the REP template title to handle this.

File: 0000-rep-template.md
Comment (shenglol) [reply 0000-rep-template.md]: Fixed.

File: README.md
Comment (shenglol) [reply README.md]: Improved this section and added a few more examples.

File: README.md
Comment (shenglol) [reply README.md]: Yeah makes sense. Incorporated that into this section.

File: 0000-rep-template.md
Comment (shenglol) [reply 0000-rep-template.md:51]: Added an Out of scope section.

File: README.md
Comment (shenglol) [reply README.md:42]: Yeah, added a blockquote to mention that.

File: 0000-rep-template.md
Comment (shenglol) [reply 0000-rep-template.md:67]: Good suggestion!

**Review by anthony-c-martin — APPROVED** [review]

**Pull request merged**
